haystack/nodes/doc_language_classifier/base.py

```diff
@@ -0,0 +1,137 @@
+import logging
+from abc import abstractmethod
+from typing import Dict, List, Optional, Tuple, Any
+
+from haystack.nodes.base import BaseComponent, Document
+
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_LANGUAGES = ["en", "de", "es", "cs", "nl"]
+
+
+class BaseDocumentLanguageClassifier(BaseComponent):
+    """
+    Abstract class for Document Language Classifiers
+    """
+
+    outgoing_edges = len(DEFAULT_LANGUAGES)
+
+    @classmethod
+    def _calculate_outgoing_edges(cls, component_params: Dict[str, Any]) -> int:
+        route_by_language = component_params.get("route_by_language", True)
+        if route_by_language is False:
+            return 1
+        languages_to_route = component_params.get("languages_to_route", DEFAULT_LANGUAGES)
+        return len(languages_to_route)
+
+    def __init__(self, route_by_language: bool = True, languages_to_route: Optional[List[str]] = None):
+        """
+        :param route_by_language: whether to send Documents on a different output edge depending on their language.
+        :param languages_to_route: list of languages, each corresponding to a different output edge (ISO code, see [langdetect` documentation](https://github.com/Mimino666/langdetect#languages)).
+        """
+        super().__init__()
+
+        if languages_to_route is None:
+            languages_to_route = DEFAULT_LANGUAGES
+            if route_by_language is True:
+                logging.warning(
```

Comment:
I would move this one down to `INFO`: important, but people that is ok with the default list should not be spammed